x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -361,48 +368,75 @@ export const getPrepopulatedEndpointException = ({
   const { file, host } = alertEcsData;
   const filePath = file?.path ?? '';
   const sha256Hash = file?.hash?.sha256 ?? '';
-  const filePathDefault = host?.os?.family === 'linux' ? 'file.path' : 'file.path.caseless';
+  const isLinux = host?.os?.name === 'Linux';
+
+  const entriesToAdd = () => {
+    if (isLinux) {
+      return addIdToEntries([
+        {
+          field: 'file.path',
+          operator: 'included',
+          type: 'match',
+          value: filePath ?? '',
+        },
+        {
+          field: 'file.hash.sha256',
+          operator: 'included',
+          type: 'match',
+          value: sha256Hash ?? '',
+        },
+        {
+          field: 'event.code',
+          operator: 'included',
+          type: 'match',
+          value: eventCode ?? '',
```

Comment:
can we dedupe a little here to show what's common?

looks like `file.hash.sha256` and `event.code` are OS-independent. And then file path is only case-sensitive for linux (maybe `field: 'file.path' + (!isLinux ? '.caseless' : '')` or something of that sort).

and then file.Ext.code_sig can be added for non-linux.

Reply:
@pzl I tried what you're suggesting before but got a pretty nasty type error.  See this commit diff - https://github.com/elastic/kibana/pull/103038/commits/aaf285d1b52614bc31067aede46dc81cab14d2e7

I'll revisit and see if there's another way

Reply:
alright, I deduped a bit - https://github.com/elastic/kibana/pull/103038/commits/680c2f1946ed2ab7994939471699703c4ccc1742